workspace/__lib__/xod/gpio/analog-read/patch.cpp

```diff
@@ -0,0 +1,28 @@
+struct State {
+    int configuredPort = -1;
+};
+
+{{ GENERATED_CODE }}
+
+void evaluate(Context ctx) {
+    if (!isInputDirty<input_UPD>(ctx))
+        return;
+
+    const uint8_t port = getValue<input_PORT>(ctx);
+    bool err = (port < A0 || port > A0 + NUM_ANALOG_INPUTS - 1);
+    if (err) {
+        emitValue<output_ERR>(ctx, 1);
+        return;
+    }
+
+    State* state = getState(ctx);
+    if (port != state->configuredPort) {
+        ::pinMode(port, INPUT);
```

Comment:
Now, since the semantics changed, do no checks. Just call `pinMode` unconditionally before each `analogRead`.

The `State` is no longer necessary